### PR TITLE
Enhance: Function and Data Type Access Specifier, and Structures using FB_Init args

### PR DESCRIPTION
**Pull request by engineerjoe440**
# Changes:
* Add support for Access Specifier for Functions
  image
* Add support for Access Specifier in Struct/Union/Enum Types
  image
* Add support for Use of Instantiated function-blocks with FB_Init arguments
  image

### Closing Thoughts:
I'm not absolutely sure that I'm following the correct practice in regards to the addition of `FunctionCall` in the `StructureElementDeclaration` [see here](https://github.com/klauer/blark/compare/master...engineerjoe440:enhancement/func-access_struct-fb-init-args_data-type-access?expand=1#diff-c26321b40313af0a101f2abfe0076eb50f721b233e65e3f5db676c2cbd4054f3R1185). That said, I'm happy to make any adjustments as you see fit! I'm sorry that I couldn't find better references for the use of access specifiers for functions and type declarations. It seems that feature is not very widely documented.

### CODE REVIEW

**Comment by klauer** [comment]
Surface-level I think this is looking good - thanks as always for the contribution!

I'd like to do a bit of a deeper dive before merging, though. Bear with me, though, I'm not sure I'll have the time today.

**Comment by klauer** [comment]
Surface-level I think this is looking good - thanks as always for the contribution!

I'd like to do a bit of a deeper dive before merging, though. Bear with me, though, I'm not sure I'll have the time today.

**Comment by engineerjoe440** [comment]
Please, don't worry about this not "happening today." I'm very appreciative for your incredible responsiveness thus far! Thank you!

I completely understand wanting to go through this a bit more. Please let me know what questions you have! I want to make sure you feel comfortable with any of my contributions!

**Comment by klauer** [comment]
I had a bit of time this morning and found there were a few things that could be simplified. In the grammar, `function_call` and `fb_invocation` are grammatically identical, but there have always been two classes to represent each of them. 

From context alone, I don't think it's possible to differentiate between a function call and a function block call (feel free to correct me if I'm wrong). So to that end, I pushed a commit up to a separate branch [32780d9](https://github.com/klauer/blark/commit/32780d9c2b385153222ae116bdbe4dfc8681a29f) that removes `fb_invocation` and replaces it with `function_call`.

There are a couple minor fixes in there along with some reminders about why those properties are there.

I also ran the test suite against more local PLC code repositories, so I think this all should be good. So if that looks/sounds good to you, cherry-pick to your branch and we can move forward with this. 👍

**Comment by engineerjoe440** [comment]
OK! I think that I've pulled that in correctly, but please let me know if there's anything I've missed. Thank you!

**Comment by klauer** [comment]
Looks good! Going to click that green merge button.

**Review by klauer — APPROVED** [review]

**Pull request merged**
